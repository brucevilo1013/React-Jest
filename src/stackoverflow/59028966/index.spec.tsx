const { OrderProgress } = require("./");
const mod = require("./");
import React from "react";
import { mount } from "enzyme";
import { act } from "react-dom/test-utils";

jest.useFakeTimers();

afterEach(() => {
  jest.restoreAllMocks();
});

describe("OrderProgress", () => {
  it("should passs", () => {
    const getRndIntegerSpy = jest
      .spyOn(mod, "getRndInteger")
      .mockReturnValueOnce(200);
    const subscribeSpy = jest.spyOn(mod.subject, "subscribe");

    const mProps = { orderData: { price: 1000 } };
    const wrapper = mount(<OrderProgress {...mProps}></OrderProgress>);
    expect(wrapper.find(".progressBar").prop("style")).toEqual({
      width: "10%",
      maxWidth: "99%"
    });
    expect(wrapper).toMatchSnapshot();
    act(() => {
      jest.advanceTimersByTime(10 * 1000);
    });
    wrapper.update();
    expect(wrapper.find(".progressBar").prop("style")).toEqual({
      width: "20%",
      maxWidth: "99%"
    });

    expect(subscribeSpy).toBeCalledTimes(1);
    expect(getRndIntegerSpy).toBeCalledWith(0, 120);
    expect(wrapper).toMatchSnapshot();
  });
});

describe("getRndInteger", () => {
  it("should pass", () => {
    const randomSpy = jest.spyOn(Math, "random").mockReturnValueOnce(1);
    const actual = mod.getRndInteger(1, 100);
    expect(actual).toBe(101);
    expect(randomSpy).toBeCalledTimes(1);
  });
});
