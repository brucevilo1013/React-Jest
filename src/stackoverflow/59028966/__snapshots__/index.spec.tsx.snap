// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderProgress should passs 1`] = `
<OrderProgress
  orderData={
    Object {
      "price": 1000,
    }
  }
>
  <div
    className="progressCard"
  >
    <div
      className="progressBar"
      style={
        Object {
          "maxWidth": "99%",
          "width": "10%",
        }
      }
    />
    <div
      className="liveData"
    >
      <span
        className="label"
      >
        LIVE
      </span>
      <span
        className="value"
      >
        100
      </span>
    </div>
    <div
      className="progressDetail"
    >
      : Awaiting Execution…
    </div>
    <div
      className="targetData"
    >
      <span
        className="label"
      >
        Target
      </span>
      <span
        className="value"
      >
        1000
      </span>
    </div>
  </div>
</OrderProgress>
`;

exports[`OrderProgress should passs 2`] = `
<OrderProgress
  orderData={
    Object {
      "price": 1000,
    }
  }
>
  <div
    className="progressCard"
  >
    <div
      className="progressBar"
      style={
        Object {
          "maxWidth": "99%",
          "width": "20%",
        }
      }
    />
    <div
      className="liveData"
    >
      <span
        className="label"
      >
        LIVE
      </span>
      <span
        className="value"
      >
        200
      </span>
    </div>
    <div
      className="progressDetail"
    >
      : Awaiting Execution…
    </div>
    <div
      className="targetData"
    >
      <span
        className="label"
      >
        Target
      </span>
      <span
        className="value"
      >
        1000
      </span>
    </div>
  </div>
</OrderProgress>
`;
